functions/src/modules/SQR.functions.ts

```diff
@@ -112,124 +112,140 @@ export const sendEmailOnNewAllotment = functions.database
+    audioFileStatus = 'audioProblem';
 
-export const processSubmissions = functions.database
-  .ref('/webforms/sqr/{submission_id}')
-  .onCreate(async (snapshot, context) => {
-    const original = snapshot.val();
-
-    let audioFileStatus = 'WIP';
-
-    if (original.not_preferred_language) audioFileStatus = 'Spare';
-    else if (original.unable_to_play_or_download)
-      audioFileStatus = 'Audio Problem';
-
-    // 1. Add the webform data to a SQR submissions DB path
-    let submission = {
-      fileName: original.audio_file_name,
-      cancellation: {
-        notPreferredLanguage: audioFileStatus === 'Spare',
-        audioProblem: audioFileStatus === 'Audio Problem',
-      },
-      soundQualityRating: original.sound_quality_rating,
-      unwantedParts: original.unwanted_parts,
-      soundIssues: original.sound_issues,
-      duration: {
-        beginning: original.beginning,
-        ending: original.ending,
-      },
-      comments: original.comments,
-      token: original.token,
-      created: original.created,
-      //  timestamp of the submission creation,
-      // can differ from COMPLETED in case of saving a DRAFT and completing later.
-      completed: original.completed, // timestamp of the submission completion.
-      changed: original.changed, //timestamp of the submission update.
-      devotee: {
-        name: original.name,
-        emailAddress: original.email_address,
-      },
-    };
-    db.ref(`/sqr/submissions/${original.serial}`).update(submission);
-
-    // 2. Update the allotment ( first get the previous NOTES )
-    const filesSnapshot = await db
-      .ref(`/sqr/files/${original.list}/${original.audio_file_name}`)
-      .once('value');
-    let allotmentUpdates = { status: audioFileStatus };
+    //  EXTRACTING the list name first from the file_name
+  let list = helpers.extractListFromFilename(submission.fileName);
+  
+
+  // 2. Update the allotment ( first get the previous NOTES )
+  const filesSnapshot = await db
+    .ref(`/sqr/files/${list}/${submission.fileName}`)
+    .once('value');
+    
+    if (filesSnapshot.exists()) {
+      let fileUpdate = { status: audioFileStatus };
+ 
+  
     // in case 1 & 2 add the comments to the notes
-    if (audioFileStatus !== 'WIP')
-      allotmentUpdates['notes'] = `${filesSnapshot.val().notes}\n${
-        original.comments
+    if (audioFileStatus !== 'WIP') {
+      fileUpdate['notes'] = `${filesSnapshot.val().notes}\n${
+        submission.comments
       }`;
-    // if the audio has a problem then REMOVE the devotee from the file allotment
-    if (audioFileStatus === 'audioProblem') allotmentUpdates['devotee'] = {};
-    db.ref(`/sqr/files/${original.list}/${original.audio_file_name}`).update(
-      allotmentUpdates
+    }
+  
+    // if the audio has any cancellation then REMOVE the assignee from the file allotment
+    if (audioFileStatus === 'audioProblem' || audioFileStatus === 'spare') 
```

Comment:
Don't check the status here, check the cancellation from the submission. This will be more obvious and clear.